File: Common/Sound/src/common/util/AudioBuffer.java
package common.util;

import javafx.scene.media.Media;
import javafx.scene.media.MediaException;

import java.io.File;
import java.io.FileNotFoundException;
import java.net.URI;
import java.net.URL;
import java.util.HashMap;

public final class AudioBuffer {

    private static final HashMap<String, Media> fileToMedia = new HashMap<>();

    private static void addFile( final String filepath ) throws MediaException {
        try {
            fileToMedia.put(
                    filepath,
                    newMedia( filepath )
                           );
        }
        catch (NullPointerException | FileNotFoundException npe ) {
            npe.printStackTrace();
        }
    }

    private static Media newMedia( final String fileName ) throws MediaException, FileNotFoundException {
        URL  url       = AudioBuffer.class.getResource(fileName);
        if (url == null) {
            throw new FileNotFoundException(fileName);
        }
        File soundFile = new File( url.getPath() );
        if (soundFile.exists()) {
            return new Media( soundFile.toURI().toString() );
        }
        throw new FileNotFoundException(fileName);
    }

    private static boolean contains( final String fileName ) {
        return fileToMedia.containsKey( fileName );
    }

    public static Media loadMedia( final String fileName ) throws MediaException {
        if ( !AudioBuffer.contains( fileName ) ) {
            AudioBuffer.addFile( fileName );
        }
        return fileToMedia.get( fileName );
    }

}

File: Common/Sound/src/common/util/AudioBuffer.kt
package common.util

import javafx.scene.media.Media
import javafx.scene.media.MediaException
import java.io.File
import java.io.FileNotFoundException
import java.util.*

object AudioBuffer {
    private val fileToMedia = HashMap<String, Media>()

    @Throws(MediaException::class, FileNotFoundException::class)
    private fun newMedia(fileName: String): Media {
        val url = AudioBuffer::class.java.getResource(fileName)
                  ?: throw FileNotFoundException(fileName)
        val soundFile = File(url.path)
        if (soundFile.exists()) {
            return Media(soundFile.toURI().toString())
        }
        throw FileNotFoundException(fileName)
    }

    @Throws(MediaException::class, FileNotFoundException::class)
    fun loadMedia(fileName: String): Media? {
        if (!fileToMedia.containsKey(fileName)) {
            fileToMedia[fileName] = newMedia(fileName)
        }
        return fileToMedia[fileName]
    }
}
